packages/fxa-payments-server/src/routes/Subscriptions/Reactivate/ConfirmationDialog.tsx

```diff
@@ -10,33 +10,52 @@ import DialogMessage from '../../../components/DialogMessage';
 import fpnImage from '../../../images/fpn';
 import { Plan, Customer } from '../../../store/types';
 import { metadataFromPlan } from 'fxa-shared/subscriptions/metadata';
+import { apiInvoicePreview } from '../../../lib/apiClient';
+import { WebSubscription } from 'fxa-shared/subscriptions/types';
+import LoadingSpinner from '../../../components/LoadingSpinner';
+import { useHandleConfirmationDialog } from '../../../lib/hooks';
 
-export default ({
-  onDismiss,
+const ConfirmationDialogErrorContent = () => (
+  <>
+    <Localized id="general-error-heading">
+      <h4 data-testid="reactivate-confirm-error-loading">
+        General application error
+      </h4>
+    </Localized>
+    <Localized id="basic-error-message">
+      <p>Something went wrong. Please try again later.</p>
+    </Localized>
+  </>
+);
+
+const ConfirmationDialogContent = ({
   onConfirm,
-  plan,
-  customer,
   periodEndDate,
+  currency,
+  productName,
+  amount,
+  last4,
+  webIconBackground,
+  webIconURL,
 }: {
-  onDismiss: Function;
   onConfirm: () => void;
-  plan: Plan;
-  customer: Customer;
   periodEndDate: number;
+  currency: string;
+  productName: string;
+  amount: number;
+  last4: string | undefined;
+  webIconBackground: string | null | undefined;
```

Comment:
Not saying this is not the case but more asking out of slight astonishment but can this really be `string | null | undefined`?

Reply:
Haha yup, apparently. I didn't look at this too much, since this is mainly to accommodate existing code that I didn't want to change.